### PR TITLE
test: Disable unreliable test on travis+ASAN_UBSAN

### PR DESCRIPTION
**Pull request by justinmk**

### CODE REVIEW

File: test/functional/ui/mouse_spec.lua [hunk 1]
```diff
@@ -153,7 +153,8 @@ describe('Mouse input', function()
     end)
 
     it('in tabline to the left moves tab left', function()
-      if os.getenv("TRAVIS") and helpers.os_name() == "osx" then
+      if os.getenv("TRAVIS") and (helpers.os_name() == "osx"
```
Comment (jamessan) [inline test/functional/ui/mouse_spec.lua]: How about having a general way to disable flaky checks and then have .travis.yml export an environment variable under these conditions to cause the flaky tests to be disabled?  That would also allow me to disable the flaky tests on some of the more touchy Debian buildds.

File: test/functional/ui/mouse_spec.lua [hunk 2]
```diff
@@ -153,7 +153,8 @@ describe('Mouse input', function()
...
+          or os.getenv("CLANG_SANITIZER") == "ASAN_UBSAN") then
         pending("[Fails on Travis macOS. #4874]", function() end)
```
Comment (jamessan) [inline test/functional/ui/mouse_spec.lua]: The message is no longer accurate.

**Comment by justinmk** [comment]
> How about having a general way to disable flaky checks and then have .travis.yml export an environment variable under these conditions to cause the flaky tests to be disabled? That would also allow me to disable the flaky tests on some of the more touchy Debian builds.

@jamessan Travis defines `CONTINUOUS_INTEGRATION` and also `CI`. Currently we check `TRAVIS` in 5 different places, but they all have different qualifiers, e.g. one of them checks the gcc version:

    if os.getenv("TRAVIS") and os.getenv("CC") == "gcc-4.9"

If we add e.g. a `SKIP_UNRELIABLE_TESTS` flag, it won't have the granularity of the current checks. Which tests are failing in the debian builds? Is it different per environment?

**Comment by jamessan** [comment]
> If we add e.g. a SKIP_UNRELIABLE_TESTS flag, it won't have the granularity of the current checks.

Can't that be covered by the test matrix?  Just export the variable in the right parts.

As far as where things have failed in the latest upload:

Arch | Failed test | Reason | Succeeded before
-------|--------------|-----------|-------------------------
[armel](https://buildd.debian.org/status/fetch.php?pkg=neovim&arch=armel&ver=0.1.7-1&stamp=1480741742) | :global is interrupted by mapped CTRL-C | mismatch in screen output | ✔️ 
[armhf](https://buildd.debian.org/status/fetch.php?pkg=neovim&arch=armhf&ver=0.1.7-1&stamp=1480738937) | :global is interrupted by mapped CTRL-C | failed to allocate memory | ✔️ 
[mips](https://buildd.debian.org/status/fetch.php?pkg=neovim&arch=mips&ver=0.1.7-1&stamp=1480739253) | jobs running tty-test program calling jobclose() | should be fixed by this PR | ✔️ 
[mips64el](https://buildd.debian.org/status/fetch.php?pkg=neovim&arch=mips64el&ver=0.1.7-1&stamp=1480739496) | :global is interrupted by mapped CTRL-C | mismatch in screen output | ✔️ 
[alpha](https://buildd.debian.org/status/fetch.php?pkg=neovim&arch=alpha&ver=0.1.7-1&stamp=1480739682) | :global is interrupted by mapped CTRL-C | mismatch in screen output | ✔️ 
[hppa](https://buildd.debian.org/status/fetch.php?pkg=neovim&arch=hppa&ver=0.1.7-1&stamp=1480746751) | :global is interrupted by mapped CTRL-C | mismatch in screen output | ✔️ 
[powerpc](https://buildd.debian.org/status/fetch.php?pkg=neovim&arch=powerpc&ver=0.1.7-1&stamp=1480738505) | printf() accepts prefix and width modifiers with %b | out of memory | ❌ 
[sparc64](https://buildd.debian.org/status/fetch.php?pkg=neovim&arch=sparc64&ver=0.1.7-1&stamp=1480774340) | unclear what was running | E: Build killed with signal TERM after 600 minutes of inactivity | ✔️ 
[x32](https://buildd.debian.org/status/fetch.php?pkg=neovim&arch=x32&ver=0.1.7-1&stamp=1480737821) | string() function used to represent dictionaries errors when dumping recursive dictionaries | out of memory | ❌

**Comment by justinmk** [comment]
The `:global` test in particular is just not a good approach, I'll fix that next.

**Comment by justinmk** [comment]
> Can't that be covered by the test matrix? Just export the variable in the right parts.

But there are some memory-bound tests here that we don't want to disable on travis. How many variables are we going to export?

**Pull request merged**